src/index.js

```diff
@@ -278,19 +294,12 @@ class Tracing {
       return null;
     }
 
-    // XXX: no empty string here v
-    // We should send the span name too
-    // TODO: take a look for span name here: https://github.com/openzipkin/zipkin-go-opentracing/blob/594640b9ef7e5c994e8d9499359d693c032d738c/propagation_ot.go#L26
-    const span = new this._Span("", {
-      traceId: {
-        traceId: carrier[HttpHeaders.TraceId],
-        parentId: carrier[HttpHeaders.ParentSpanId],
-        spanId: carrier[HttpHeaders.SpanId],
-        sampled: carrier[HttpHeaders.Sampled]
-      }
+    return new TraceId({
```

Comment:
Yeah, might be better to have some duplication here for extracting. Although I wonder here if this doesn't break a contract. I think extract should return a span rather then just a traceId, shouldn't it?

Reply:
that is the API/contract `extract()` is defining: https://github.com/opentracing/opentracing-javascript/blob/d2c1c6a5a50439bd769d02c7f4638293db40140a/src/tracer.ts#L132-L156

Reply:
But wouldn't it return a `span context`: https://github.com/opentracing/opentracing-javascript/blob/d2c1c6a5a50439bd769d02c7f4638293db40140a/src/tracer.ts#L178 

This is what we call span in this project and not a TraceID, so we break the contract here I would say